// @docusaurus/plugin-ideal-image/lib/index.js
'use strict';
/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
Object.defineProperty(exports, '__esModule', {value: true});
const tslib_1 = require('tslib');
const path_1 = (0, tslib_1.__importDefault)(require('path'));
function default_1(_context, options) {
  return {
    name: 'docusaurus-plugin-ideal-image',
    getThemePath() {
      return path_1.default.resolve(__dirname, './theme');
    },
    configureWebpack(_config, isServer) {
      if (process.env.NODE_ENV !== 'production') {
        return {};
      }
      return {
        mergeStrategy: {
          'module.rules': 'prepend',
        },
        module: {
          rules: [
            {
              test: /\.(png|jpe?g|gif)$/i,
              use: [
                require.resolve('@docusaurus/lqip-loader'),
                {
                  loader: require.resolve('@docusaurus/responsive-loader'),
                  options: {
                    emitFile: !isServer,
                    // eslint-disable-next-line global-require
                    adapter: require('@docusaurus/responsive-loader/sharp'),
                    name: 'assets/ideal-img/[name].[hash:hex:7].[width].[ext]',
                    ...options,
                  },
                },
              ],
            },
          ],
        },
      };
    },
  };
}
exports.default = default_1;
